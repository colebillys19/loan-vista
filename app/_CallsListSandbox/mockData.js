/* eslint-disable prettier/prettier */
const MOCK_DATA = [
  {
    audio: 'deeder.mp3',
    date: '05/13/2021',
    dept: 'ABC',
    desc: 'bibendum imper magna yeet',
    id: 'cb50e5c5-fed2-43da-8a47-fc8f7df96222',
    rep: 'DEF',
    time: '06:00 AM',
  },
  {
    audio: 'hoydo.mp3',
    date: '05/14/2021',
    dept: 'GHI',
    desc: 'yo yoo yooo yoooo yooo yoo yo yoo yooo yoooo yooo yoo yo yoo yooo yoooo yooo yoo yo yoo yooo yoooo yooo yoo yo yoo yooo yoooo yooo yoo yo yoo yooo yoooo yooo yoo yo yoo yooo yoooo yooo yoo yo yoo yooo yoooo yooo yoo',
    id: 'cb50e5c5-fed2-43da-8a47-fc8f7df96223',
    rep: 'JKL',
    time: '07:00 AM',
  },
  {
    audio: 'shtude.mp3',
    date: '05/15/2021',
    dept: 'MNO',
    desc: 'magna bibendum imper amet',
    id: 'cb50e5c5-fed2-43da-8a47-fc8f7df96224',
    rep: 'PQR',
    time: '08:00 AM',
  },
  {
    audio: 'shtude.mp3',
    date: '05/15/2021',
    dept: 'MNO',
    desc: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
    id: 'cb50e5c5-fed2-43da-8a47-fc8fadf96224',
    rep: 'PQR',
    time: '08:00 AM',
  },
  {
    audio: 'shtude.mp3',
    date: '05/15/2021',
    dept: 'MNO',
    desc: 'magna bibendum imper amet',
    id: 'bb50e5c5-fed2-43da-8a47-fc8fadf96224',
    rep: 'PQR',
    time: '08:00 AM',
  },
];

export default MOCK_DATA;
