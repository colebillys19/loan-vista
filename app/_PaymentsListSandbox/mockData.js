const MOCK_DATA = [
  {
    accidentHealthInsurance: '$66.86',
    accruedLateCharge: '$70.42',
    date: '12/30/2020',
    desc: 'Principal Curtailment',
    effectiveDate: '01/02/2020',
    escrow: '$41.02',
    escrowBalance: '$38.82',
    fees: '$74.24',
    id: '29ba8a70-19d3-4ddf-b4ff-879b4f8c8fe6',
    interest: '$25.00',
    lifeInsurance: '$55.66',
    misc: '$45.28',
    otherTotal: '$11.32',
    principal: '$36.02',
    principalBalance: '$8.68',
    replacementReserve: '$97.76',
    restrictedReserve: '$15.48',
    suspense: '$83.46',
    time: '04:12 AM',
    total: '$14.93',
  },
  {
    accidentHealthInsurance: '$78.39',
    accruedLateCharge: '$26.40',
    date: '12/29/2020',
    desc: 'Principal Curtailment',
    effectiveDate: '01/02/2020',
    escrow: '$89.24',
    escrowBalance: '$10.64',
    fees: '$63.92',
    id: '748a3cec-9c60-4dd5-9979-5e70feaf7093',
    interest: '$54.94',
    lifeInsurance: '$20.55',
    misc: '$25.65',
    otherTotal: '$12.47',
    principal: '$41.33',
    principalBalance: '$54.80',
    replacementReserve: '$29.82',
    restrictedReserve: '$7.42',
    suspense: '$91.42',
    time: '08:16 PM',
    total: '$34.89',
  },
  {
    accidentHealthInsurance: '$31.63',
    accruedLateCharge: '$38.14',
    date: '12/26/2020',
    desc: 'Mortgage Payment',
    effectiveDate: '01/02/2020',
    escrow: '$12.79',
    escrowBalance: '$9.06',
    fees: '$17.95',
    id: 'fb74124b-3e78-4ab8-8178-10d18cf28b30',
    interest: '$47.38',
    lifeInsurance: '$15.12',
    misc: '$10.87',
    otherTotal: '$92.89',
    principal: '$62.03',
    principalBalance: '$15.99',
    replacementReserve: '$19.95',
    restrictedReserve: '$20.48',
    suspense: '$13.12',
    time: '09:32 PM',
    total: '$15.66',
  },
  {
    accidentHealthInsurance: '$39.74',
    accruedLateCharge: '$69.90',
    date: '12/25/2020',
    desc: 'Initial Escrow Deposit',
    effectiveDate: '01/02/2020',
    escrow: '$99.98',
    escrowBalance: '$46.00',
    fees: '$45.46',
    id: '152f341c-bba1-4813-aa53-26979a5dc80b',
    interest: '$75.10',
    lifeInsurance: '$74.43',
    misc: '$27.71',
    otherTotal: '$61.84',
    principal: '$46.51',
    principalBalance: '$49.87',
    replacementReserve: '$18.26',
    restrictedReserve: '$72.42',
    suspense: '$44.91',
    time: '05:53 PM',
    total: '$5.66',
  },
  {
    accidentHealthInsurance: '$90.40',
    accruedLateCharge: '$48.37',
    date: '12/16/2020',
    desc: 'Principal Curtailment',
    effectiveDate: '01/02/2020',
    escrow: '$40.58',
    escrowBalance: '$21.33',
    fees: '$3.24',
    id: '2d15a80f-1d71-49ff-90e1-82246574a4de',
    interest: '$74.84',
    lifeInsurance: '$77.91',
    misc: '$77.54',
    otherTotal: '$83.57',
    principal: '$77.68',
    principalBalance: '$29.92',
    replacementReserve: '$65.43',
    restrictedReserve: '$45.81',
    suspense: '$58.61',
    time: '05:57 AM',
    total: '$72.89',
  },
];

export default MOCK_DATA;
