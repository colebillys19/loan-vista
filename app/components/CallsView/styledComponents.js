import styled from 'styled-components';

import { StyledTableHead } from 'components/_base-ui/GradientListTable';
import LinkButton from 'components/_base-ui/LinkButton';

export const BaseTableData = styled.td`
  padding: 0;
  position: relative;
`;

export const CustomTableHead = styled(StyledTableHead)`
  & th:nth-of-type(1) {
    width: 11.4rem;
  }
  & th:nth-of-type(2) {
    width: 10.2rem;
  }
  & th:nth-of-type(3) {
    width: 7.6rem;
  }
  & th:nth-of-type(4) {
    width: 6.8rem;
  }
  & th:nth-of-type(6) {
    width: 5.6rem;
  }
`;

export const StyledLinkButton = styled(LinkButton)`
  position: absolute;
  right: 0;
  top: 0.4rem;
`;
