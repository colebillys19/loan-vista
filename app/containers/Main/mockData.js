export const sidebarLoanSummaryData = [
  { label: 'Loan Type', value: 'FHA Residential' },
  { label: 'Status', value: 'Current' },
  { label: 'Principal Balance', value: '$234,456.78' },
  { label: 'Original Loan Amount', value: '$345,567,89' },
  { label: 'Investor Name', value: 'Nevada Housing' },
  { label: 'Investor Code', value: 'N03' },
  { label: 'Assumption Code', value: 'A' },
  { label: 'Assumption Date', value: '2/3/2020' },
  { label: 'Interest Rate', value: '5%' },
  { label: 'Origination Date', value: '5/6/2012' },
  { label: 'Maturity Date', value: '5/6/2042' },
  { label: 'Grace Days', value: '15' },
  { label: 'Lien Position', value: '1st Mortgage' },
];
