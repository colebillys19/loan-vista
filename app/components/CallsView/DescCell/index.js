import React, { useLayoutEffect, useRef, useState } from 'react';
import T from 'prop-types';

import ConditionalRender from 'components/_shared/ConditionalRender';

import ExpandableCell from './ExpandableCell';
import { DescTableData } from './styledComponents';

const DescCell = ({ children }) => {
  const cellRef = useRef(null);
  const [renderExpandable, setRenderExpandable] = useState(false);

  useLayoutEffect(() => {
    const {
      current: { clientHeight },
    } = cellRef;
    if (clientHeight > 30) {
      setRenderExpandable(true);
    }
  }, [cellRef]);

  return (
    <ConditionalRender
      Component={<ExpandableCell>{children}</ExpandableCell>}
      FallbackComponent={
        <DescTableData ref={cellRef}>{children}</DescTableData>
      }
      shouldRender={renderExpandable}
    />
  );
};

DescCell.propTypes = { children: T.node.isRequired };

export default DescCell;
