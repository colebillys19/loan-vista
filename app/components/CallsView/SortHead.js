/**
 * TableHead
 * @description ...
 */

import React from 'react';
import T from 'prop-types';

import ListSortButton from 'components/_shared/ListSortButton';
import {
  StyledTableHeader,
  StyledTableRow,
} from 'components/_shared/GradientListTable';

import { CallsTableHead } from './styledComponents';

const SortHead = ({
  dispatchFetchCallsData,
  headers,
  lastSortCol,
  lastSortOrder,
  sortLoading,
}) => {
  const handleSortClick = (header) => {
    const newSortCol = header;
    const newSortOrder =
      header === lastSortCol && lastSortOrder === 'desc' ? 'asc' : 'desc';

    dispatchFetchCallsData(newSortCol, newSortOrder);
  };

  return (
    <CallsTableHead>
      <StyledTableRow>
        {headers.map((header) => {
          if (['date', 'dept', 'rep'].includes(header)) {
            return (
              <StyledTableHeader key={header} scope="col">
                <ListSortButton
                  isActive={header === lastSortCol}
                  isAscending={
                    header === lastSortCol && lastSortOrder === 'asc'
                  }
                  loading={sortLoading === header}
                  onClick={() => handleSortClick(header)}
                  text={header}
                />
              </StyledTableHeader>
            );
          }

          return (
            <StyledTableHeader key={header} scope="col">
              {header}
            </StyledTableHeader>
          );
        })}
      </StyledTableRow>
    </CallsTableHead>
  );
};

SortHead.propTypes = {
  dispatchFetchCallsData: T.func.isRequired,
  headers: T.arrayOf(T.string).isRequired,
  lastSortCol: T.string.isRequired,
  lastSortOrder: T.string.isRequired,
  sortLoading: T.oneOfType([T.bool, T.string]).isRequired,
};

export default SortHead;
