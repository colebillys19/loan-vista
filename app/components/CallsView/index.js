/**
 * CallsView
 * @description ...
 */

import React from 'react';
import T from 'prop-types';

import CallsList from 'components/CallsList';
import TabWrapper from 'components/TabWrapper';
import ConditionalRender from 'components/_baseUI/ConditionalRender';
import Spinner from 'components/_baseUI/Spinner';
import NoListDataFallback from 'components/_baseUI/NoListDataFallback';

import { CallsViewWrapper } from './styledComponents';

const CallsView = ({ callsData, loading }) => {
  const noListData = !loading && callsData.length === 0;

  return (
    <TabWrapper aria-labelledby="calls-tab" id="calls-view">
      <CallsViewWrapper>
        <ConditionalRender
          Component={
            <ConditionalRender
              Component={<CallsList callsData={callsData} loading={loading} />}
              FallbackComponent={<Spinner />}
              shouldRender={!loading}
            />
          }
          FallbackComponent={<NoListDataFallback />}
          shouldRender={!noListData}
        />
      </CallsViewWrapper>
    </TabWrapper>
  );
};

CallsView.propTypes = {
  callsData: T.array.isRequired,
  loading: T.bool.isRequired,
};

export default CallsView;
