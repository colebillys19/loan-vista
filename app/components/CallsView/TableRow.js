import React from 'react';
import T from 'prop-types';

import {
  StyledTableData,
  StyledTableRow,
} from 'components/_shared/GradientListTable';
import { BaseTableData, StyledLinkButton } from './styledComponents';

import DescCell from './DescCell';

const TableRow = ({ data, headers }) => (
  <StyledTableRow>
    {headers.map((header) => {
      if (header === 'desc') {
        return <DescCell key={header}>{data[header]}</DescCell>;
      }

      if (header === 'audio') {
        return (
          <BaseTableData key="audio">
            <StyledLinkButton onClick={() => null} text="request" />
          </BaseTableData>
        );
      }

      return <StyledTableData key={header}>{data[header]}</StyledTableData>;
    })}
  </StyledTableRow>
);

TableRow.propTypes = {
  data: T.shape({
    audio: T.string,
    date: T.string,
    dept: T.string,
    desc: T.string,
    rep: T.string,
    time: T.string,
  }),
  headers: T.arrayOf(T.string).isRequired,
};

export default TableRow;
