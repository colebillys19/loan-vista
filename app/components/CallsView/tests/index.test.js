import React from 'react';
import { render } from 'react-testing-library';

import CallsView from '../index';

import { MOCK_DATA } from './mockData';

const mockProps = { callsData: MOCK_DATA, loading: false };

const Component = <CallsView {...mockProps} />;

describe('<CallsView />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(Component);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(Component);
    expect(firstChild).toMatchSnapshot();
  });
});
