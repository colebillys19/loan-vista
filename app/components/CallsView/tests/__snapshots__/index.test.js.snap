// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CallsView /> Should render and match the snapshot 1`] = `
<section
  aria-labelledby="calls-tab"
  class="TabContainer-sc-1us62xt-0 geMfQv"
  id="calls-view"
>
  <form
    class="styledComponents__StyledForm-sc-6nnwf7-2 kpDAEG"
  >
    <span
      class="styledComponents__DisableWrapper-sc-6nnwf7-0 fNPNLa"
    >
      <label
        class="styledComponents__StyledLabel-sc-6nnwf7-4 fnnyyM"
        for=""
      >
        from
      </label>
      <span
        class="styledComponents__DatePickerWrapper-sc-1wk5y4r-1 sdcXD"
      >
        <div
          class="MuiFormControl-root MuiTextField-root styledComponents__StyledKeyboardDatePicker-sc-1wk5y4r-2 doVNdr styledComponents__StyledTextField-sc-1wk5y4r-3 eubNBY"
          title=""
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
              placeholder="date"
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          
        </div>
        <div
          class="styledComponents__ErrorIndicator-sc-1wk5y4r-0 oDsFE"
        />
      </span>
    </span>
    <span
      class="styledComponents__DisableWrapper-sc-6nnwf7-0 fNPNLa"
    >
      <label
        class="styledComponents__StyledLabel-sc-6nnwf7-4 fnnyyM"
        for=""
      >
        to
      </label>
      <span
        class="styledComponents__DatePickerWrapper-sc-1wk5y4r-1 sdcXD"
      >
        <div
          class="MuiFormControl-root MuiTextField-root styledComponents__StyledKeyboardDatePicker-sc-1wk5y4r-2 doVNdr styledComponents__StyledTextField-sc-1wk5y4r-3 eubNBY"
          title=""
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
              placeholder="date"
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          
        </div>
        <div
          class="styledComponents__ErrorIndicator-sc-1wk5y4r-0 oDsFE"
        />
      </span>
    </span>
    <span
      class="styledComponents__DisableWrapper-sc-6nnwf7-0 fNPNLa"
    >
      <label
        class="styledComponents__StyledLabel-sc-6nnwf7-4 fnnyyM"
        for=""
      >
        or
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline styledComponents__StyledSelect-sc-6nnwf7-6 dzbarT"
      >
        <div
          aria-haspopup="listbox"
          class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          role="button"
          tabindex="0"
        >
          date range
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput"
          tabindex="-1"
          value="0"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </span>
    <span
      class="styledComponents__DisableWrapper-sc-6nnwf7-0 fNPNLa"
    >
      <label
        class="styledComponents__StyledLabel-sc-6nnwf7-4 fnnyyM"
        for=""
      >
        and/or
      </label>
      <div
        class="MuiFormControl-root MuiTextField-root styledComponents__StyledTextField-sc-6nnwf7-7 bukyEO"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            placeholder="keyword search"
            type="text"
            value=""
          />
        </div>
      </div>
    </span>
    <button
      aria-label="delete"
      class="MuiButtonBase-root MuiIconButton-root styledComponents__StyledIconButton-sc-6nnwf7-3 LvLiT"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          style="height: 2.5rem; width: 2.5rem;"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="transparent"
          />
          <path
            d="M17.65 6.35c-1.63-1.63-3.94-2.57-6.48-2.31-3.67.37-6.69 3.35-7.1 7.02C3.52 15.91 7.27 20 12 20c3.19 0 5.93-1.87 7.21-4.56.32-.67-.16-1.44-.9-1.44-.37 0-.72.2-.88.53-1.13 2.43-3.84 3.97-6.8 3.31-2.22-.49-4.01-2.3-4.48-4.52C5.31 9.44 8.26 6 12 6c1.66 0 3.14.69 4.22 1.78l-1.51 1.51c-.63.63-.19 1.71.7 1.71H19c.55 0 1-.45 1-1V6.41c0-.89-1.08-1.34-1.71-.71l-.64.65z"
            fill="#7b68ee"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <span
      class="styledComponents__DisableWrapper-sc-6nnwf7-0 fNPNLa"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text styledComponents__StyledButton-sc-6nnwf7-1 fyzjVE styledComponents__StyledButton-sc-86ulmm-0 exBHac"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          find
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </span>
  </form>
  <div
    class="styledComponents__TableContainer-vlp0pm-7 jqujHF"
  >
    <div
      class="GradientBorder-sc-1rv7gqq-0 hyonyR"
    />
    <div
      class="GradientBorder-sc-1rv7gqq-0 dfrFlw"
    />
    <div
      class="GradientBorder-sc-1rv7gqq-0 fJnNtj"
    />
    <div
      class="GradientBorder-sc-1rv7gqq-0 dmTzmF"
    />
    <table
      class="styledComponents__StyledTable-vlp0pm-1 ecXuWn"
    >
      <thead
        class="styledComponents__StyledTableHead-vlp0pm-4 styledComponents__CustomTableHead-dnzlrp-0 ifSOoJ"
      >
        <tr
          class="styledComponents__StyledTableRow-vlp0pm-6 eXzxfo"
        >
          <th
            class="styledComponents__StyledTableHeader-vlp0pm-5 jzAiFL"
            scope="col"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text styledComponents__StyledButton-sc-1qurbx9-0 fqrnhO MuiButton-disableElevation"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                date
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    style="height: 0.8rem; width: 0.8rem;"
                    viewBox="7 7 10 10"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M7 10l5 5 5-5z"
                      fill="#888888"
                    />
                  </svg>
                </span>
              </span>
            </button>
          </th>
          <th
            class="styledComponents__StyledTableHeader-vlp0pm-5 jzAiFL"
            scope="col"
          >
            time
          </th>
          <th
            class="styledComponents__StyledTableHeader-vlp0pm-5 jzAiFL"
            scope="col"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text styledComponents__StyledButton-sc-1qurbx9-0 fqrnhO MuiButton-disableElevation"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                dept
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    style="height: 0.8rem; width: 0.8rem;"
                    viewBox="7 7 10 10"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M7 10l5 5 5-5z"
                      fill="#888888"
                    />
                  </svg>
                </span>
              </span>
            </button>
          </th>
          <th
            class="styledComponents__StyledTableHeader-vlp0pm-5 jzAiFL"
            scope="col"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text styledComponents__StyledButton-sc-1qurbx9-0 fqrnhO MuiButton-disableElevation"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                rep
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    style="height: 0.8rem; width: 0.8rem;"
                    viewBox="7 7 10 10"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M7 10l5 5 5-5z"
                      fill="#888888"
                    />
                  </svg>
                </span>
              </span>
            </button>
          </th>
          <th
            class="styledComponents__StyledTableHeader-vlp0pm-5 jzAiFL"
            scope="col"
          >
            desc
          </th>
          <th
            class="styledComponents__StyledTableHeader-vlp0pm-5 jzAiFL"
            scope="col"
          >
            audio
          </th>
        </tr>
      </thead>
      <tbody
        class="styledComponents__StyledTableBody-vlp0pm-2 hzYCgB"
      >
        <tr
          class="styledComponents__StyledTableRow-vlp0pm-6 eXzxfo"
        >
          <td
            class="styledComponents__StyledTableData-vlp0pm-3 fqIKzF"
          >
            date
          </td>
          <td
            class="styledComponents__StyledTableData-vlp0pm-3 fqIKzF"
          >
            time
          </td>
          <td
            class="styledComponents__StyledTableData-vlp0pm-3 fqIKzF"
          >
            dept
          </td>
          <td
            class="styledComponents__StyledTableData-vlp0pm-3 fqIKzF"
          >
            rep
          </td>
          <td
            class="styledComponents__StyledTableData-vlp0pm-3 fqIKzF"
          >
            desc
          </td>
          <td
            class="styledComponents__StyledTableData-vlp0pm-3 fqIKzF"
          >
            <button
              class="MuiButtonBase-root styledComponents__StyledButton-sc-9o47wt-0 fgJPGq"
              tabindex="0"
              type="button"
            >
              request
            </button>
          </td>
        </tr>
      </tbody>
      <thead>
        <tr>
          <td
            class="styledComponents__TableFooterCell-vlp0pm-8 dEwaCH"
          >
            hidden text
          </td>
        </tr>
      </thead>
    </table>
  </div>
</section>
`;
